scripts/nzedb_pre_import.py

```diff
@@ -78,82 +96,143 @@ def processNzedbPre():
+		except:
+			print("Pre-Import: Error downloading/unzipping. Please try again.")
+			exit(0)
 
 
+	if fileExists:
+		dirtyChunk = pandas.read_table('predb_dump-062714.csv', sep='\t', header=None, na_values='\\N', usecols=[0,8,10,14,16,18,20,22], names=COLNAMES, chunksize=10000, engine='c', error_bad_lines=False)
+	else:
+		print("Pre-Import: File predb_dump-062714.csv not found, please try again.")	
+		exit(0)
 
-			#Query to find any existing pres, we need to delete them so COPY doesn't fail
-			with db_session() as db:
-				pres = db.query(Pre).filter(Pre.name.in_(names)).all()
 
-				prenamelist = []
-				for pre in pres:
-					prenamelist.append(pre.name)
-				
-				#Create the inverse list, basically contains pres that already exist
-				newdata = data[~data['name'].isin(prenamelist)]
-				
-				newdata.to_csv('formattedUL.csv', index=False, header=False)
+	i = 0
+	for chunk in dirtyChunk: 
+		process(chunk)
+		print("Pre-Import: Imported chunk {}".format(i))
+		i += 1
 
-				#Delete any pres found as we are essentially going to update them
-				if len(pres) is not 0:
-					for pre in pres:
-						db.delete(pre)
-				db.commit()
 
-			#Process the now clean CSV
-			conn = engine.raw_connection()
-			cur = conn.cursor()		
-			formattedUL = open('formattedUL.csv')	
+def process(precsv, processingFile=None):
+		
+	ordering = ['name','filename','nuked','category','pretime','source','requestid','requestgroup','searchname']
+	
+	#Clean up the file a bit.
+	precsv.replace("'", '', inplace=True, regex=True)
+	precsv["nuked"].replace("2", "0", inplace=True)
+	precsv["nuked"].replace("3", "1", inplace=True)
+	precsv["nuked"].replace("4", "1", inplace=True)
+	precsv["nuked"].replace("5", "1", inplace=True)
+	precsv["nuked"].replace("69", "0", inplace=True)
+	precsv.replace(".\\N$", '', inplace=True, regex=True)
+
+	#Sometimes there are duplicates within the table itself, remove them
+	precsv.drop_duplicates(subset='name', take_last=True, inplace=True)
+
+	#Add clean searchname column
+	precsv['searchname'] = precsv['name'].map(lambda name: releases.clean_release_name(name))
+
+	#Drop the pres without requestid's
+	precsv = precsv[precsv.requestid != '0'] 		
+	
+	#Create a list of names to check if they exist
+	names = list(precsv.name)
+
+	#Query to find any existing pres, we need to delete them so COPY doesn't fail
+	with db_session() as db:
+		
+		if names is not None:
+			pres = db.query(Pre).filter(Pre.name.in_(names)).all()
+
+			prenamelist = []
+			for pre in pres:
+				prenamelist.append(pre.name)
+		
+		#Create the inverse list, basically contains pres that already exist
+		newdata = precsv[~precsv['name'].isin(prenamelist)]
+
+		data = io.StringIO()
+		precsv.to_csv(data, index=False, header=False)
+		
+		#Delete any pres found as we are essentially going to update them
+		if len(newdata) is not 0:
+			for pre in pres:
+				db.delete(pre)
+			db.commit()
+		else:
+			print("Pre-Import: No pres to add from this file")
+		

```

Comment:
(I wrote all of this then realized I may have found the issue!)

@Murodese Can you sanity check this? As this is the area where it messes up on the delete/insert.

169: I remove duplicates from the file itself, picking up the last row. Basically means if something was nuked/unuked we get the most recent value.

175: Removes any pres without requestids

178-88: Creates the list of names we can use in the in query on 184. These are then added to a list so that on 191 we can get the inverse and delete them.

190: Create an inverse of the list so that we can delete the results. Actually, this is where the error is I think. I don't want the inverse, I just want to delete the ones where the names are in the list. Let me try that